Merging js parse

jsarabia opened the pull request.

Commit:
fix casing for resources and createapp sql

mbr4477 committed:
Update to get collection by slug and container id

Commit:
Nullable resource metadata

temporary fix for inserting bible and testament without a resource container backing them

Commit:
importing chapters/verses

mbr4477 commented on src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/database/AppDatabase.kt:
```diff
@@ -30,7 +30,7 @@ class AppDatabase(
 
         // Check if the database file exists
-        val schemaFile = File(listOf("src", "main", "Resources", "CreateAppDb.sql").joinToString(File.separator))
+        val schemaFile = File(listOf("src", "main", "resources", "CreateAppDb.sql").joinToString(File.separator))
```
I know this isn't relevant to your changes on this branch, but I realized yesterday that this should be changed to something more like: `File(ClassLoader.getSystemResource("CreateAppDb.sql").toURI())`

jsarabia replied:
yeah, for sure

mbr4477 commented on src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/entities/CollectionEntity.kt:
```diff
@@ -8,5 +8,5 @@ data class CollectionEntity(
         var slug: String,
         var sort: Int,
-        var metadataFk: Int
+        var metadataFk: Int?
```
Are we keeping this as nullable for now?

jsarabia replied:
yeahhhhhhhh unfortunately

Commit:
Synchronized inserts, nullable resource container field

Commit:
Get db sql from system resource rather than hardcoded path

Commit:
Improve the sqldirpath variable

projectDir.path seemed to be needed for working with cli gradle at least on the rpi
